File: TOMIMSV4/Entity/Transacciones/Picking/Trans_Picking_Det_Parametros/clsBeTrans_picking_det_parametros.vb
Public Class clsBeTrans_picking_det_parametros
    Implements ICloneable
    Implements IDisposable

    Public Property IdParametroPicking() As Integer = 0
    Public Property IdPickingDet() As Integer = 0
    Public Property IdProductoParametro() As Integer = 0
    Public Property Valor_texto() As String = ""
    Public Property Valor_numerico() As Nullable(Of Double)
    Public Property Valor_fecha() As Nullable(Of Date)
    Public Property Valor_logico() As Nullable(Of Boolean)
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now

    Sub New()
    End Sub

    Sub New(ByRef IdParametroPicking As Integer, ByVal IdPickingDet As Integer, ByVal IdProductoParametro As Integer, ByVal valor_texto As String, ByVal valor_numerico As Double, ByVal valor_fecha As Date, ByVal valor_logico As Boolean, ByVal user_agr As String, ByVal fec_agr As Date)
        Me.IdParametroPicking = IdParametroPicking
        Me.IdPickingDet = IdPickingDet
        Me.IdProductoParametro = IdProductoParametro
        Me.Valor_texto = Valor_texto
        Me.Valor_numerico = Valor_numerico
        Me.Valor_fecha = Valor_fecha
        Me.Valor_logico = Valor_logico
        Me.User_agr = User_agr
        Me.Fec_agr = Fec_agr
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Picking/clsBeTrans_picking_det_parametros.cs
namespace WMS.EntityCore.Picking
{
    public class clsBeTrans_picking_det_parametros : ICloneable, IDisposable
    {
        public int IdParametroPicking { get; set; } = 0;
        public int IdPickingDet { get; set; } = 0;
        public int IdProductoParametro { get; set; } = 0;
        public string Valor_texto { get; set; } = "";
        public double? Valor_numerico { get; set; }
        public DateTime? Valor_fecha { get; set; }
        public bool? Valor_logico { get; set; }
        public string User_agr { get; set; } = "";
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public bool IsNew { get; set; } = false;

        public clsBeTrans_picking_det_parametros()
        {
        }

        public clsBeTrans_picking_det_parametros(int IdParametroPicking, int IdPickingDet, int IdProductoParametro, string valor_texto, double? valor_numerico, DateTime? valor_fecha, bool? valor_logico, string user_agr, DateTime fec_agr)
        {
            this.IdParametroPicking = IdParametroPicking;
            this.IdPickingDet = IdPickingDet;
            this.IdProductoParametro = IdProductoParametro;
            this.Valor_texto = valor_texto;
            this.Valor_numerico = valor_numerico;
            this.Valor_fecha = valor_fecha;
            this.Valor_logico = valor_logico;
            this.User_agr = user_agr;
            this.Fec_agr = fec_agr;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        public void Dispose()
        {
            // Dispose implementation if needed
        }
    }
}
